### PR TITLE
Resume threads if deallocated when in background

### PR DESCRIPTION
**Pull request by kingst**
I wasn't able to trigger this bug myself, but I did see a report from Crashlytics on the user study and I'm 90% sure that this fixes it.

### CODE REVIEW

**Comment by jaimejiyepark** [comment]
shouldnt the mlqueue resuming and suspension be in the mutexqueue as well to ensure they behave properly before it backgrounds or not?

File: CardScan/Classes/CreditCardOcr/OcrMainLoop.swift [hunk 1]
```diff
@@ -260,7 +260,7 @@ public class OcrMainLoop : MachineLearningLoop {
     }
     
     @objc func didBecomeActive() {
-        inBackground = false
+        mutexQueue.sync { self.inBackground = false }
         for queue in machineLearningQueues {
```
Comment (jaimejiyepark) [inline CardScan/Classes/CreditCardOcr/OcrMainLoop.swift:264]: should this be in the mutexQueue?

Comment (kingst) [reply CardScan/Classes/CreditCardOcr/OcrMainLoop.swift:264]: it's just to make sure that all shared state is accessed from the `mutexQueue` note that we're using `sync` here

**Comment by kingst** [comment]
The sync call on the machine learning queues make sure that the ML is done before the thread is suspended

**Comment by jaimejiyepark** [comment]
but the for loop isnt in the mutex queue. shouldnt it be in the mutex queue?

**Comment by jaimejiyepark** [comment]
#shipit

**Pull request merged**